tsdb/tsdbblockutil.go

```diff
@@ -61,30 +39,35 @@ func CreateBlock(samples []*MetricSample, dir string, mint, maxt int64, logger l
 	if chunkRange < 0 {
 		return "", ErrInvalidTimes
 	}
-	chunkDir := filepath.Join(dir, "chunks_tmp")
-	defer func() {
-		os.RemoveAll(chunkDir)
-	}()
-	head, err := CreateHead(samples, chunkRange, chunkDir, logger)
+
+	w := NewTSDBWriter(logger, dir)
+	err := w.initHead(chunkRange)
 	if err != nil {
 		return "", err
 	}
-	defer head.Close()
 
-	compactor, err := NewLeveledCompactor(context.Background(), nil, logger, ExponentialBlockRanges(DefaultBlockDuration, 3, 5), nil)
+	ctx := context.Background()
+	app := w.Appender(ctx)
+
+	for _, sample := range samples {
+		_, err = app.Add(sample.Labels, sample.TimestampMs, sample.Value)
+		if err != nil {
+			return "", err
+		}
+	}
+	err = app.Commit()
 	if err != nil {
 		return "", err
 	}
 
-	err = os.MkdirAll(dir, 0777)
+	ulid, err := w.Flush(ctx)
 	if err != nil {
 		return "", err
 	}
 
-	ulid, err := compactor.Write(dir, head, mint, maxt, nil)
+	err = w.Close()
```

Comment:
We should have this in a `defer` after initialising `w` to avoid not cleaning up the files in case of any errors.